predicators/structs.py

```diff
@@ -1548,6 +1584,8 @@ def __str__(self) -> str:
 
 
 # Convenience higher-order types useful throughout the code
+Observation = Any
+GoalDescription = Any
```

Comment:
Why are these type Any, instead of something like Tuple[Dict[Object, Array], Optional[Any]] and Set[GroundAtom]?

Reply:
we want observations and goal descriptions to be unconstrained -- they will often be the types that you mentioned, but observations could be images or other sensory stuff, and goal descriptions could be natural language or universally quantified formulas or something else